hibernate-core/src/main/java/org/hibernate/query/internal/NativeQueryImpl.java

```diff
@@ -559,6 +567,33 @@ protected boolean isNativeQuery() {
 		return this;
 	}
 
+	@Override
+	protected boolean applyQuerySpaces(Object value) {
+		if ( value == null ) {
+			return false;
+		}
+
+		if ( value instanceof List ) {
+			this.querySpaces.addAll( (List) value );
```

Comment:
I suspect `querySpaces` might not be initialized here. All other writing methods check for it, and initialize it with `new ArrayList<>();`.

Secondarily, but why is it using a list and not a Set? Is a List really preferreable here?

Reply:
I chose List because it is more conceptually similar to array.  We can certainly accept both.

Checking for null was just an oversight

Reply:
> Checking for null was just an oversight

WDYM, don't you think it is necessary here?

Reply:
Or do you mean for the field?  For the field, yes, that's actually not at all new and its done that way because it more closely matches array, which is what the code that uses this actually expects ultimately

Reply:
Oversight means it was overlooked.  I just missed it

Reply:
np :) I'm just wondering if you're planning to fix that.

No need to change the List/Set - that question was an unrelated curiosity, I was just wondering